homeassistant/components/mqtt/climate.py

```diff
@@ -606,14 +666,46 @@ def handle_temperature_low_received(msg: ReceiveMessage) -> None:
         @log_messages(self.hass, self.entity_id)
         def handle_temperature_high_received(msg: ReceiveMessage) -> None:
             """Handle target temperature high coming via MQTT."""
-            handle_temperature_received(
+            handle_climate_attribute_received(
                 msg, CONF_TEMP_HIGH_STATE_TEMPLATE, "_attr_target_temperature_high"
             )
 
         add_subscription(
             topics, CONF_TEMP_HIGH_STATE_TOPIC, handle_temperature_high_received
         )
 
+        @callback
+        @log_messages(self.hass, self.entity_id)
+        def handle_current_humidity_received(msg: ReceiveMessage) -> None:
+            """Handle current humidity coming via MQTT."""
+            handle_climate_attribute_received(
+                msg, CONF_CURRENT_HUMIDITY_TEMPLATE, "_attr_current_humidity"
+            )
+
+        add_subscription(
+            topics, CONF_CURRENT_HUMIDITY_TOPIC, handle_current_humidity_received
+        )
+
+        @callback
+        @log_messages(self.hass, self.entity_id)
+        def handle_target_humidity_received(msg: ReceiveMessage) -> None:
+            """Handle target humidity coming via MQTT."""
+
+            if not self.supported_features & ClimateEntityFeature.TARGET_HUMIDITY:
```

Comment:
You need some additional tests to ensure code test coverage here.

Reply:
I added a test for `handle_target_humidity_received`, but it only tests the happy path. I don't think that it's possible for `handle_target_humidity_received` to be called without having the `TARGET_HUMIDITY` feature ever since I added this validation check: 

https://github.com/Stonos/core/blob/81ca8dcb1779ac2a8f92c3f9e51831e849492eb5/homeassistant/components/mqtt/climate.py#L247-L257

Perhaps we should remove the check from `handle_target_humidity_received`?

Reply:
Sound good to me!